packages/extension-chakra-storefront/src/types/config.ts

```diff
@@ -45,22 +46,80 @@ type EinsteinAPI = {
     isProduction: boolean
 }
 
+type ShippingCountry = {
+    value: string
+    label: string
+}
+
 // Indicates where a value should be placed in the URL.
 type UrlPlacement = 'path' | 'query_string' | 'none'
 
-type Pages = typeof import('../pages')
+// Default configuration type
+// should we keep string | string[] type here??
+type DefaultPageConfig = {
+    path: string
+}
 
+type CustomPageConfigs = {
+    Account: DefaultPageConfig & {
+        orderSearchParam: {
+            limit: number
+            offset: number
+            sort: string
+            refine: []
+        }
+    }
+    Checkout: DefaultPageConfig & {
+        shippingCountryCode: ShippingCountry[]
+    }
+    Home: DefaultPageConfig & {
+        productLimit?: number
+        mainCategory?: string
```

Comment:
Should these config be required instead of optional?

Reply:
Since this is for HomePage, It is likely the page will be change for different customer, they can choose to no have the feature section which means, these config are not required.